frontend/public/components/monitoring/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,493 @@
+        isSubFormValid = !_.isEmpty(pagerDutyValues.integrationKey);
+        break;
+      case 'webhook':
+        isSubFormValid = !_.isEmpty(webhookUrl);
+        break;
+      default:
+        isSubFormValid = false;
+    }
+
+    return !_.isEmpty(receiverType) && !_.isEmpty(receiverName) && isSubFormValid;
+  };
+
+  return (
+    <div className="co-m-pane__body">
+      <Helmet>
+        <title>{titleVerb} Receiver</title>
+      </Helmet>
+      <form className="co-m-pane__body-group co-m-pane__form" onSubmit={save}>
+        <h1 className="co-m-pane__heading">Create {saveAsDefaultReceiver && 'Default'} Receiver</h1>
+        <div className="form-group">
+          <label className="control-label co-required">Receiver Name</label>
+          <input
+            className="pf-c-form-control"
+            type="text"
+            value={receiverName}
+            onChange={(e) => setReceiverName(e.target.value)}
+            aria-describedby="receiver-name-help"
+            name="receiverName"
+            id="receiver-name"
+            required
+          />
+        </div>
+        <div className="form-group co-m-pane__dropdown">
+          <label className="control-label co-required">Receiver Type</label>
+          <Dropdown
+            title={'Select Receiver Type...'}
+            items={receiverTypes}
+            dropDownClassName="dropdown--full-width"
+            id="receiver-type"
+            selectedKey={receiverType}
+            onChange={setReceiverType}
+          />
+        </div>
+
+        {receiverType === 'pagerduty' && (
+          <PagerDutySubForm
+            pagerDutyValues={pagerDutyValues}
+            setPagerDutyValues={setPagerDutyValues}
+          />
+        )}
+        {receiverType === 'webhook' && (
+          <WebhookSubForm webhookUrl={webhookUrl} setWebhookUrl={setWebhookUrl} />
+        )}
+
+        {receiverType && (
+          <RoutingLabels
+            routeLabels={routeLabels}
+            setRouteLabels={setRouteLabels}
+            saveAsDefaultReceiver={saveAsDefaultReceiver}
+          />
+        )}
+
+        <ButtonBar errorMessage={errorMsg} inProgress={inProgress}>
+          <ActionGroup className="pf-c-form">
+            <Button type="submit" variant="primary" id="save-changes" isDisabled={!isFormValid()}>
+              {saveButtonText}
+            </Button>
+            <Button type="button" variant="secondary" id="cancel" onClick={history.goBack}>
+              Cancel
+            </Button>
+          </ActionGroup>
+        </ButtonBar>
+      </form>
+    </div>
+  );
+};
+
+export const ReceiverWrapper: React.FC<ReceiverFormsWrapperProps> = React.memo(
+  ({ obj, ...props }) => {
+    const [inProgress, setInProgress] = React.useState(true);
+
+    React.useEffect(() => {
+      if (inProgress && !_.isEmpty(obj.data)) {
+        setInProgress(false);
+      }
+    }, [inProgress, obj.data]);
+
+    if (inProgress) {
+      return <LoadingBox />;
```

Comment:
I think this will cause errors displayed by `StatusBox` (`403` error, `404` error, etc.) to not be shown if `obj.data` is empty. That would be solved by using `loaded` instead of `inProgress`.

Reply:
Ok, fixed in all alert manager 'wrappers' (main config page, forms, and yaml editor).